lecture_1/homework/app/controllers/samurais_controller.rb

```diff
@@ -0,0 +1,52 @@
+class SamuraisController < ApplicationController
+  before_action :set_clan
+  before_action :set_clan_samurai, only: [:show, :update, :destroy]
+
+  # GET /clans/:clan_id/samurais
+  def index
+    json_response(@clan.samurais)
+  end
+
+  # GET /clans/:clan_id/samurais/:id
+  def show
+    json_response(@samurai)
+  end
+
+  # POST /clans/:clan_id/samurais
+  def create
+    @clan.samurais.create!(samurai_params)
+    json_response(@clan, :created)
+  end
+
+  # PUT /clans/:clan_id/samurais/:id
+  def update
+    @samurai.update(samurai_params)
+  end
+
+  # DELETE /clans/:clan_id/samurais/:id
+  def destroy
+    @samurai.destroy
+  end
+
+  def dead
```

Comment:
Najlepiej nie tworzyć takich `nierestowych` akcji, tylko obsługiwać je jakoś w odpowiedniej restowej akcji(dostępne w CRUD to show, index, create, update, destroy) albo wydzielić do nowego zagnieżdżonego kontrolera, w naszym przypadku 100% full super restowo ścieżki by wyglądały tak:

```ruby
resources :clans, param: :name  do
  resources :samurais do
    resources :alive, only: %i[index]
    resources :dead, only: %i[index]
  end
end
```

Oczywiście staramy się unikać zagnieżdżeń więc to póki co samo obsługiwanie 2 query params w indexie by wystarczyło, ale jeśli mielibyśmy np. inne akcje `update` dla żywych i martwych samurai, to taką architekturę dużo łatwiej się rozbudowuje i ładniej dzieli tą logikę :D